### ISSUE TITLE
[Feature Request] Add option to disable notification dialogues.

### ISSUE THREAD

**Issue by EveningStarNM** [gpodder/gpodder#485]
Maybe I've missed it somewhere, but I've looked at all of the settings in Preferenses and config, and I can't find a way to turn off those dialogues that pop up every time gpodder has completed downloading a file. I'm running gpodder on Windows 10 Enterprise, and I've hidden it in my system tray (using RBTray) because I want it to run silently in the background. do it's thing, and not bother me when I'm doing something else. (I've also run gpodder occasionally on Linux Mint,, but I'm giving up on that since it's even less configurable there: it can't be reduced to an out-of-the-way icon off the main part of the taskbar.)

Is there already a setting to do that? If not, I'd be very grateful to get one.

**Comment by Marrib** [comment]
I totally agree, this download complete dialog/notification popup  is very annoying. Indeed, there should be an option in the preferences to disable that notification.

**Comment by bob46859** [comment]
It'd be great to make the synchronization failure dialog optional too.  It popping up and stealing focus is pretty aggravating.

**Comment by Marrib** [comment]
Another approach to the notifications:
Instead of popping up in a notification dialog, notifications could appear in a notification bar at the top or bottom of gPodder or in an appearing notification section in the header
![untitled-1](https://user-images.githubusercontent.com/40357209/41864236-84b78848-78a9-11e8-8ed4-29ca1b1a4959.jpg)
. Here's a mockup:

**Comment by Rcola1987** [comment]
A question to people having this issue. Which i am having too. Was Gpodder simply failing silently in previous versions because i'm getting about 20 to 30% of my feeds are getting errors and these very aggravating popups. In my case after a full sync i have to close over 300 of the Finished/Failed dialog boxes! Please add option to disable and simply Log the errors in a reasonable place.

**Comment by EveningStarNM** [comment]
That's an interesting question, @Rcola1987 . I may not remember correctly, but in version 9.something, I think that I remember seeing failure notices somewhere in the window frame, where they should be, instead of in popups. But I may not remember correctly. I didn't pay much attention to my RSS downloader. I didn't want to. If it failed to download a Defensive Security or DTNS podcast, I really didn't care. I looked at it once a day to load up Winamp (I don't care what anyone says, I love my ancient, pre-reorganization version of Winamp, and I hope it works forever) and then forgot about it.

**Comment by jskrdla** [comment]
I'd love to see an option to silence these as well. I keep getting popups throughout the day. The periodic stealing focus, popping up messages during meetings, and having to close 30+ messages each morning when I start the day - it is rather annoying.

**Comment by elelay** [comment]
Remove [line 43 in notification-win32.py](https://github.com/gpodder/gpodder/blob/master/share/gpodder/extensions/notification-win32.py#L43) and you won't get notifications anymore. You will then be able to enable the extension again in Preferences > Extensions > Desktop Integration > Notification bubbles for windows.

@Marrib thanks for the constructive input. A notification zone is also described in [this stackoverflow](https://stackoverflow.com/a/45437257) but is not trivial to implement.

**Comment by Marrib** [comment]
@elelay Thanks for regarding user's feedback. I wish more devs would be so open to comments like the gPodder folks are.

Your solution worked. Maybe this could become a regular visible option in gPodder preferences so that users don't have to edit that py-file beforehand.

I can totally understand that more complex notification features take a lot of effort to implement. Honestly, I don't really need these super nifty kinds of notifications, it was just a design idea I found interesting to share. I can live in total peace with the given way to disable the notifications.  So again, thanks for sharing that solution.

And keep up the good work. I really think that gPodder (at least on Windows) is the only podcast client that's really usable at this time. iTunes is too bloated and other podcatchers are either outdated or not as functional.

**Comment by EveningStarNM** [comment]
@elelay THANK YOU! I tried commenting out the line instead of deleting it, but that wasn't sufficient. When I followed your instructions and deleted the line, it worked exactly as you described. I'm VERY grateful. And, yes, I don't doubt that coding for the notification zone is non-trivial. I read the comments in the file. Perhaps next year. For now, gpodder is behaving the way I want it to.

Thank you, again!

**Comment by jskrdla** [comment]
Thanks so much @elelay  - I've always been too nervous to dive into Python myself, but your fix worked perfectly! Thank you again!

**Comment by MichaelApproved** [comment]
It's not working for me either with Windows 7 with 3.10.3. The notifications are still showing up after removing the line.

**Comment by elelay** [comment]
Try removing the file completely and a file named notification-win32.py**c** if it exists.

**Comment by brrd** [comment]
This solution looks like a monkey patch. IMHO it would be much better to have a real option in gpodder instead.

@EveningStarNM Can we please reopen this issue?

**Comment by abraxas1410** [comment]
> Try removing the file completely and a file named notification-win32.pyc if it exists.

Sorry, but this doesn't work either.

What about notification.pyc and notification.py? Should I try to remove them?

**Comment by elelay** [comment]
You still see notifications after removing notification-win32.py?!
Have you restarted the program?
If so, please, also restart the computer when it's convenient for you, to be sure (and no, you don't have to pull the plug and check your internet connection is working. It's just that I don't see how notifications can show up for you when the code for it is removed. Sounds like magic ;-) )...

**Comment by abraxas1410** [comment]
I am sorry, it's magic. I did restart the computer and I still see the popups.

**Comment by omoknen** [comment]
I am having the same problem as abraxas1410.  I did remove the file as well and restarted.  I have to click on three dialog boxes before files will download.

**Comment by elelay** [comment]
Where were the files you removed and where is the executable (gpodder.exe) you launch?
Thanks,

**Comment by elelay** [comment]
I'm at loss to what happens there. Here is a way to find out:
Use [procmon](https://docs.microsoft.com/en-us/sysinternals/downloads/procmon) to confirm where the notification-win32.py code comes from?
1. run procmon
2. add a filter to include if process name is gpodder.exe
![procmon_filter](https://user-images.githubusercontent.com/222469/43285196-7d03915e-911e-11e8-8b47-6567f690fa5b.png)
3. run D:\Applications\Winapp\gPodder\bin\gpodder.exe, download a podcast to have the popup show up.
4. in procmon.exe, save to file in CSV and attach it to a comment.
Thanks,

**Comment by omoknen** [comment]
Okay thank you.  I have problem reading so this might be a little rough.
Try one:  The first file is procmon without the file notification-win32.py one podcast.  It shows a blank dialog box with an okay button.  I waited and selected a manual update of a podcast, and **I get no dialog box**, and the file loads on it 
Try two:  The second file is procmon without the file notification-win32.py automatic get all podcasts.  Sorry the file is large I have a lot of podcasts. 
 I turned back on check on startup for all my podcasts, and after it has checked.  **I get a dialog box.**  I have to select the button before it will download.  I check for podcasts every hour and I think that I was always getting the dialog box.

[Logs.zip](https://github.com/gpodder/gpodder/files/2233763/Logs.zip)

**Comment by elelay** [comment]
@omoknen you don't get a dialog AFTER podcasts are downloaded or when a refresh error happens? So this is not the dialogs previously discussed in this issue.
Now, the dialog box you see could be the list of new episodes to either download or ignore (3 buttons at the bottom: "Obsolete" / "Cancel" / "Download"). This dialog can be disabled in the settings: tab "Updating", select "Download Immediately" "When new episodes are found".

**Comment by elelay** [comment]
@omoknen thanks for the clarification. It would help if you could resize the dialog, so that we can see what the title is. Thanks

**Comment by omoknen** [comment]
The dialog doesn't resize unfortunately.

**Comment by elelay** [comment]
@omoknen thank you! We finally got to the bottom of the problem:
there was a bug in gPodder triggered on rare occasions (#482) causing an unexpected exception, which likely causes the dialog you see.
It is a [simple fix](https://github.com/gpodder/gpodder/commit/7d6e476435b6f6b73ac806b81454bff151df066a) (in data/lib/python3.6/site-packages/gpodder) or you can wait for next gPodder release.

**Comment by omoknen** [comment]
@elelay The fix worked - no dialog box.
Thank you for all your patience and time.

**Comment by elelay** [comment]
@omoknen great to hear

**Comment by elelay** [comment]
@EveningStarNM, @Marrib and other interested users, could you try the updated [notification-win32.py](https://raw.githubusercontent.com/gpodder/gpodder/a321db85e3ee26522edcbe5d52600cae09dcf9f0/share/gpodder/extensions/notification-win32.py)?
It should  show nice native notifications, especially on Windows 10.
I'd appreciate your confirmation that it works before releasing...

**Comment by Marrib** [comment]
With this updated version of [notification-win32.py](https://raw.githubusercontent.com/gpodder/gpodder/a321db85e3ee26522edcbe5d52600cae09dcf9f0/share/gpodder/extensions/notification-win32.py) no notification popups appear at all. With the original version included in the gPodder release 3.10.3 download notification popups appear.
I did a fresh setup of gPodder by downloading the portable Version and replacing the included .py with the updated .py. I am running Windows 8.1 x64.

**Comment by Marrib** [comment]
So I launched gpodder.exe -v from the command prompt as described in the doc. No popup notifications appear and I don't see any debug/verbose messages, neither in gPodder no in the command prompt window.

**Comment by elelay** [comment]
@Marrib it's actually **gpodder-cmd.exe -v** not gpodder.exe

**Comment by Marrib** [comment]
Sorry, my bad. 

Started "gpodder-cmd.exe -v" this time. Still no notification dialogs or popups.
Attached is a text file with the contents of the command prompt window and log file.

[2018-07-29.log](https://github.com/gpodder/gpodder/files/2238966/2018-07-29.log)

**Comment by elelay** [comment]
Ha, yes, sorry:  missing [gpodder.ico](https://raw.githubusercontent.com/gpodder/gpodder/master/tools/win_installer/misc/gpodder.ico) to be put in data/bin (path indicated in the log file).

**Comment by Marrib** [comment]
Ok, that worked. When the download of a test podcast episode finished, a small gPodder icon appeared in my system tray together with a small notification bubble telling me that the download of that episode completed.

**Comment by elelay** [comment]
Great, thank you both :-)

**Comment by abraxas1410** [comment]
I have replaced the notification-win32.py and copied gpodder.ico to the bin folder. Now I see a windows 10 notification after a download has finished. It disappears on its own or is replaced by a new one after the next download and doesn't disturb any more. Thanks for this.

I attached a debug log file just in case you'd like to have it.
[gPodder-debug-log.txt](https://github.com/gpodder/gpodder/files/2242464/gPodder-debug-log.txt)
